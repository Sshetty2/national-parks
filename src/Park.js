import React, { Component } from 'react';

class Park extends Component {
 constructor(props) {
    super(props);
    this.state = {
      displayFull: false  
   };
 }
  showFullCard= () => {
    this.setState({
      displayFull: true
    });
  }

  removeCard = (event) => {
    event.preventDefault();
    this.props.removeCard();
    this.setState({
      displayFull: false
    });
  }

  render() {
    let imagePath = `./${this.props.selectedPark.image}`;
    return (
      <div className="card-overlay">
        <div className="park-card-small"> 
          <p onClick={this.removeCard}>X</p>
          <h1 className="park-title">{this.props.selectedPark.parkName} National Park</h1>
          <img className="park-img" src={imagePath} />
          {/* {imageSource} */}
          <h3 className="park-text">{this.props.selectedPark.description.substring(0, 80)}...</h3>
          <button className="button-small" onClick={this.showFullCard}>View More</button>
          <p>Visited</p>
          <p>Bucket</p>
        </div>
      </div>
    )
  }
}

export default Park;