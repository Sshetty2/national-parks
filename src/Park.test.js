import React from 'react';
import Park from './Park';
import { shallow } from 'enzyme';

let selectedPark = {
  parkName: "Death Valley",
  state: ["California", "Nevada"],
  dateEstablished: "October 31, 1994",
  latitude: "36.24°N",
  longitude: "116.82°W",
  annualVisitors: 1294827,
  petsWelcome: true,
  editorsChoice: "Badwater Basin Salt Flats",
  websiteUrl: "https://www.nps.gov/deva/index.htm",
  urlCode: "deva",
  image: "assets/death-valley.jpg",
  imageRef: "<a rel=\"nofollow\" class=\"external text\" href=\"https://sites.google.com/site/thebrockeninglory/\">Brocken Inaglory</a> - <span class=\"int-own-work\" lang=\"en\">Own work</span>, <a href=\"https://creativecommons.org/licenses/by-sa/3.0\" title=\"Creative Commons Attribution-Share Alike 3.0\">CC BY-SA 3.0</a>, <a href=\"https://commons.wikimedia.org/w/index.php?curid=5990426\">Link</a>",
  sizeAcres: 337306314,
  description: "The Badlands are a collection of buttes, pinnacles, spires, and mixed-grass prairies. The White River Badlands contain the largest assemblage of known late Eocene and Oligocene mammal fossils. The wildlife includes bison, bighorn sheep, black-footed ferrets, and prairie dogs."
};

describe('Park', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Park
        selectedPark={selectedPark}
      />
    );
  });
  
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});