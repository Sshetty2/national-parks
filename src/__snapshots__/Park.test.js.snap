// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Park should match the snapshot with all data passed in 1`] = `
<div
  className="card-overlay"
>
  <div
    className="park-card-small"
  >
    <p
      onClick={[Function]}
    >
      X
    </p>
    <h1
      className="park-title"
    >
      Death Valley
       National Park
    </h1>
    <img
      className="park-img"
      src="./assets/death-valley.jpg"
    />
    <h3
      className="park-text"
    >
      The Badlands are a collection of buttes, pinnacles, spires, and mixed-grass prai
      ...
    </h3>
    <button
      className="button-small"
      onClick={[Function]}
    >
      View More
    </button>
    <p>
      Visited
    </p>
    <p>
      Bucket
    </p>
  </div>
</div>
`;
